### PR TITLE
Oracle DATE/TIMESTAMP/TIMESTAMP_TZ fixes

### LINKED ISSUE #431: Oracle call uses To_Timestamp not To_Date
```cs
var pollTime = DateTime.Now.AddMinutes(-15);  

var contracts = (from c in dbConnection.GetTable<Contract>()
                                     where c.Last_Modified > pollTime
                                     select c.Contract_ID);
```
The above linq creates following sql

```sql
--  Oracle.Managed Oracle
DECLARE @pollTime TimeStamp -- DateTime
SET     @pollTime = TO_TIMESTAMP('2016-09-08 10:42:31.2056108', 'YYYY-MM-DD HH24:MI:SS.FF7')

SELECT
               t1.Contract_ID
FROM
               Contract t1
WHERE
               t1.Last_Modified > :pollTime
```

### PR DESCRIPTION
**Pull request by MaceWindu**
Fix #431

### CODE REVIEW

**Comment by MaceWindu** [comment]
I'm moving issue/fix to v3, as it will not work properly without several fixes we implemented in v3

File: Tests/Linq/UserTests/Issue0431Tests.cs [hunk 1]
```diff
@@ -0,0 +1,181 @@
+﻿using LinqToDB;
+using LinqToDB.Mapping;
+using NUnit.Framework;
+using System;
+using System.Linq;
+using Tests.Model;
+
+namespace Tests.UserTests
+{
+	public class Issue0431Tests : TestBase
```
Comment (sdanyliv) [inline Tests/Linq/UserTests/Issue0431Tests.cs]: Maybe better to move this test to provider’s test. Without linking to issue.

Comment (MaceWindu) [reply Tests/Linq/UserTests/Issue0431Tests.cs]: agree

**Comment by MaceWindu** [comment]
Still don't have enough type information in our model to make it work. Will try to add it as separate pr first

**Comment by MaceWindu** [comment]
This is done.

Summary:
Reviewed date/time types support for Oracle.
- DataType.Date maps to `date` (without time). Preffered type: DateTime
- DataType.DateTime maps to `date` (with time). Preffered type: DateTime
- DataType.DateTime2 maps to `timestamp` + precision supported. Preffered type: DateTime
- DataType.DateTimeOffset maps to `timestamp with timezone` + precision supported. Preffered type: DateTimeOffset
- fixed/implemented literals generation for for all 4 types
- DateTimeOffset parameters now respect parameters inlining settings

File: Source/LinqToDB/DataProvider/Oracle/Oracle11SqlOptimizer.cs [hunk 2]
```diff
@@ -28,7 +28,7 @@ public override SqlStatement Finalize(SqlStatement statement)
 		{
 			CheckAliases(statement, 30);
 
-			new QueryVisitor().VisitAll(statement, SetQueryParameter);
+			//new QueryVisitor().VisitAll(statement, SetQueryParameter);
```
Comment (sdanyliv) [inline Source/LinqToDB/DataProvider/Oracle/Oracle11SqlOptimizer.cs]: Remove than then.

**Comment by MaceWindu** [comment]
ok, looks like azure broken this weekend. first azure bot died, now it generates sql, not posible with latest source code...

**Pull request merged**